### DISCUSSION TITLE
📣 All about contributing and triaging

### DISCUSSION THREAD

**Discussion by aeharding** [aeharding/voyager#180] [General]
## Note: This guide is now outdated

More info coming soon!

-----

Hello! Thanks for your interest an help in contributing to wefwef. The amount of interest this project has received (which started as a personal project I never thought anyone else would use) has been incredible.

I can only keep up with so much, and so I am creating this discussion to talk about current priorities and triaging process.

The growth of this project has changed its priorities. wefwef is now how a significant portion of users access and interact with the Threadiverse.

## Triaging

Current priorities in order

 1. things that [are broken](https://github.com/aeharding/wefwef/releases/tag/0.9.1) **AND** [stop the user flow](https://github.com/aeharding/wefwef/pull/160). There may not be linked issues for these depending on how critical
 2. Things that maintain the health of the threadiverse. An example is [marking new posts as NSFW](https://github.com/aeharding/wefwef/issues/150) or [reporting](https://github.com/aeharding/wefwef/issues/174)
 4. things that look broken but don't stop the user flow, and/or are annoying. [example](https://github.com/aeharding/wefwef/issues/147)
 5. Things that may happen in lemmy but the user doesn't see. An example is implementing the inbox for notifications. Exception: Mod and admin actions are coming, but not a priority at this level
 6. New features that I prioritize
 7. New features that have the most community interest

For the immediate future, I plan on taking all 1) and 2) priority items.

~~### New features that I am prioritizing over the next week~~

**This weeks work is done!** Please see #347 for next planned work

1.  [x] Swipe to hide
2. [x] Saving posts (this is also somewhat a priority under bullet point 4 above)
3. [x] iOS splash screens
4. [x] Favorites
5. [x] Top sort options
6. [x] Improved gallery
7. [x] Quick community switcher

### New features that the community prioritizes

I'm trying to determine the best way to do this. It may involve gauging interest on !wefwef@lemmy.world, as well as # of thumbs up on issues via Github. Someone willing to implement with a PR may also boost its priority somewhat.

## Stepping on toes

The most important thing is to stay in communication. Make sure there's a bug for it, and make sure I'm not assigned. If you'd like to take something, make a comment so everyone is aware. Making sure that I acknowledge you working on something will help me not be blind-sighted by it. Make sure PRs reference issues.

## Making new PRs

Speaking of making new PRs: You're welcome to (after confirming through an issue), but please be aware that it may not be merged for a while, especially if its an unprioritized new feature. There are 15 outstanding PRs just from yesterday, and I cannot review them all for a while. The project is also changing day-by-day, so conflicts may need to be resolved regularly.

At this point with this much interest in contributing, I may close PRs that I'm not expecting. Please don't take it personally. I really appreciate how passionate and exciting everyone is about contributing.

Over time things will calm down, and we'll get a better process.

## Discussion below

Let's talk below on priorities.

**Comment by IverCoder** [comment]
Thank you for this project! When can we contribute translations?

**Reply by aeharding** [reply]
Translations is definitely something to be done long term, but its too early right now :)

**Reply by Gromga** [reply]
Is there anyway to get notified when the time comes for translations ? I’d also like to contribute on that front

**Comment by geneccx** [comment]
Awesome work - thank you for building an amazing app.

One thing that really stood out to me was the server component and it acting as a proxy on behalf of the user.
I believe this is a significant security risk in doing this, as it exposes user credentials to the server.
This was quite surprising to me, considering that the app is built as a PWA. Is it possible to remove this piece and have the PWA directly talk to the instance?

Otherwise, right now everyone using the Lemmy app will appear to come from the same IP address, and may also be subject to a shared rate-limit depending on the implementation.

Would love to hear your thoughts and brainstorm some potential solutions.

**Reply by aeharding** [reply]
Ohhhh booyyy are you in for some drama

The past week has been hell because webapps are forced to proxy.

I've spent hours coordinating with admins including lemmy.world

We're waiting on this PR to be merged https://github.com/LemmyNet/lemmy/pull/3421

It can't come soon enough. I would like more than anything to stop proxying asap

**Reply by geneccx** [reply]
Appreciate the context. I'm glad it's already on your radar.

**Reply by geneccx** [reply]
Good news, this is now live for Lemmy instances on BE: 0.18.1-rc.10!

**Comment by aashu16** [comment]
Thank you for building wefwef!

Do you have any thoughts and/or plans regarding tests?

I understand that the app is changing constantly but adding e2e (Playwright/Cypress) tests might be helpful at this stage, starting with the more "complete" pages. Perhaps some unit and component tests too wherever needed.

**Reply by aeharding** [reply]
Hi there! I just replied to your email and this is definitely something I'm interested in :)

**Reply by PaulACoroneos** [reply]
Haha was about to come in here and post. I’ve got experience with jest and cypress in conjunction with testing library which I strongly suggest using. I think one of the best ways to make sure we don’t step on each others toes is to get some testing in place to prevent regressions of core functionality. 

I’m on vacation visiting my family but I really want to contribute back to OSS and this seems like a great project.
